### ISSUE TITLE
headers print twice

### ISSUE THREAD

**Issue by moofmayeda** [jdlubrano/stream_lines#127]
Got some really funky behavior when trying to read and then copy a csv with headers, where iterating over the first row included both the fields and the headers. I was able to create a failing test to show what I mean: https://github.com/jdlubrano/stream_lines/pull/126

**Comment by jdlubrano** [comment]
Hi @moofmayeda!  Nice to hear from you, and I hope that you are doing well.

Thank you for submitting the failing test, and I agree that the behavior that you are seeing is surprising.

Could you please tell me a bit more about your use case?  As much as this library tries to behave like a normal `Enumerable`, the library is still streaming data from an HTTP request.  If you call `first` and then iterate over the stream, you would actually be initiating two HTTP requests.  In theory the first request would "stop" after the first row is read (somehow...maybe...I dunno I guess it depends on the internals of Enumerable 😬 ).  The weirdness you are seeing is due to some of the internals of this library, but I am wondering what your overall use case to see if we can take a more efficient route in general.

For example, if you are strictly copy a CSV with its headers, I would recommend not parsing the CSV at all.  CSV parsing is _pretty_ slow.  Streaming the raw lines would be much, much quicker and all of the data would be copied in CSV format.

If you want to parse the CSV to somehow process data as you stream it, we might be stuck with something like:

```ruby
stream = StreamLines::Reading::CSV.new(url, headers: true)
cloud = []

stream.each_with_index do |row, i|
  cloud << row.headers.to_csv if i.zero?
  cloud << row.fields.to_csv
end
```

Admittedly the stream abstraction leaks its way into your code using this approach, but I've seen worse 😅 .  Ultimately, I see ways to "fix" the bug in this library, but I don't see a great way to avoid making two HTTP requests if you call `first` and then iterate over the stream.  Avoiding multiple HTTP requests would require storing the HTTP response in memory, which contradicts the purpose of this library as it tries to be memory efficient by reading streams of data rather than the entirety of responses.

**Comment by jdlubrano** [comment]
Hi @moofmayeda .

This bug should be fixed in version `0.4.1` of `stream_lines`.  Please also see the [caveats](https://github.com/jdlubrano/stream_lines#caveats) section about `Enumerable`s and streaming in the README to avoid duplicative network requests, though!
